### PR TITLE
chore: replace use of XORTrials with OneVarTrials in TestPyTorchTrial [MLG-42]

### PR DESCRIPTION
**Pull request by drh-determined-ai**
## Description

chore: replace use of XORTrials with OneVarTrials in TestPyTorchTrial \[[MLG-42]\]

This also affects `test_local.test_test_one_batch`.

The module `tests.experiment.fixtures.pytorch_xor_model` is now completely unused, so its file has been removed.

## Test Plan

Unit tests.

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[MLG-42]: https://determinedai.atlassian.net/browse/MLG-42?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: harness/tests/experiment/fixtures/pytorch_onevar_model.py [hunk 1]
```diff
@@ -407,6 +449,200 @@ def __init__(self, context: pytorch.PyTorchTrialContext) -> None:
         )
 
 
+class OneVarTrialCustomEval(BaseOneVarTrial):
+    _searcher_metric = "loss"
```
Comment (mpkouznetsov) [inline harness/tests/experiment/fixtures/pytorch_onevar_model.py:453]: If I am not mistaken, the name of the searcher metric refers to the metric returned by the evaluation and not by training. So this should probably be `"val_loss"` (the same name as on line 464).

File: harness/tests/experiment/fixtures/pytorch_onevar_model.py [hunk 2]
```diff
@@ -407,6 +449,200 @@ def __init__(self, context: pytorch.PyTorchTrialContext) -> None:
...
+                data, labels = data.cuda(), labels.cuda()
+            output = self.model(data)
+            loss_sum += self.loss_fn(output, labels)
+
+        loss = loss_sum / len(data_loader)
+        return {"val_loss": loss}
+
+
+class OneVarTrialAccessContext(BaseOneVarTrial):
+    _searcher_metric = "loss"
```
Comment (mpkouznetsov) [inline harness/tests/experiment/fixtures/pytorch_onevar_model.py:468]:
```suggestion
    _searcher_metric = "val_loss"
```

File: harness/tests/experiment/fixtures/pytorch_onevar_model.py [hunk 3]
```diff
@@ -407,6 +449,200 @@ def __init__(self, context: pytorch.PyTorchTrialContext) -> None:
...
+            )
+
+        else:
+            self.context.step_optimizer(self.opt)
+
+        return {"loss": loss}
+
+
+class OneVarTrialWithNonScalarValidation(BaseOneVarTrial):
+    _searcher_metric = "val_loss"
```
Comment (mpkouznetsov) [inline harness/tests/experiment/fixtures/pytorch_onevar_model.py:555]: You have to pick one of "predictions" and "mse".

**Review by mpkouznetsov — CHANGES_REQUESTED** [review]
I don't see any problems beyond metric naming. I think after fixing that this can be merged.

**Review by mpkouznetsov — APPROVED** [review]
